packages/react/dialog/src/Dialog.tsx

```diff
@@ -125,14 +128,11 @@ const DialogOverlay = forwardRefWithAs<typeof DialogOverlayImpl, DialogOverlayOw
   }
 );
 
-const DialogOverlayImpl = forwardRefWithAs<typeof OVERLAY_DEFAULT_TAG>((props, forwardedRef) => {
-  const { as: Comp = OVERLAY_DEFAULT_TAG, ...overlayProps } = props;
-  return (
-    <Portal>
-      <Comp {...getPartDataAttrObj(OVERLAY_NAME)} ref={forwardedRef} {...overlayProps} />
-    </Portal>
-  );
-});
+const DialogOverlayImpl = forwardRefWithAs<typeof Primitive>((props, forwardedRef) => (
```

Comment:
I didn't do it here because this uses `Portal` which may have its own hooks and things internally that would need to "reset" on mount/unmount (e.g. the `forceUpdate()`). 

I only removed the abstraction when it was a component that just rendered `Primitive` because those don't do anything.